hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/VolumeIOStats.java

```diff
@@ -18,121 +18,135 @@
 
 package org.apache.hadoop.ozone.container.common.volume;
 
-import java.util.concurrent.atomic.AtomicLong;
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MetricsRegistry;
+import org.apache.hadoop.metrics2.lib.MutableCounterLong;
 
 /**
  * This class is used to track Volume IO stats for each HDDS Volume.
  */
 public class VolumeIOStats {
+  private String name = VolumeIOStats.class.getSimpleName();
 
-  private final AtomicLong readBytes;
-  private final AtomicLong readOpCount;
-  private final AtomicLong writeBytes;
-  private final AtomicLong writeOpCount;
-  private final AtomicLong readTime;
-  private final AtomicLong writeTime;
+  private @Metric MutableCounterLong readBytes;
+  private @Metric MutableCounterLong readOpCount;
+  private @Metric MutableCounterLong writeBytes;
+  private @Metric MutableCounterLong writeOpCount;
+  private @Metric MutableCounterLong readTime;
+  private @Metric MutableCounterLong writeTime;
+  private MetricsRegistry registry;
 
   public VolumeIOStats() {
-    readBytes = new AtomicLong(0);
-    readOpCount = new AtomicLong(0);
-    writeBytes = new AtomicLong(0);
-    writeOpCount = new AtomicLong(0);
-    readTime = new AtomicLong(0);
-    writeTime = new AtomicLong(0);
+    init();
+  }
+
+  public VolumeIOStats(String identifier) {
+    this.name += '-' + identifier;
+    init();
+  }
+
+  public void init() {
+    DefaultMetricsSystem.initialize(name);
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.register(name, "Volume I/O Statistics", this);
```

Comment:
We should have the corresponding VolumeIOStats  unregister method to unregister these metric.

Reply:
oops. thanks for the reminder!